pad-levels.js

```diff
@@ -1,14 +1,59 @@
 /* eslint no-unused-vars: 0 */
 'use strict';
 
-const format = require('./format');
+class Padder {
+  constructor(opts = {}) {
+    if (opts.levels) {
+      this.addPadding(opts.levels, opts.filler);
+    }
+
+    this.options = opts;
+  }
+
+  static getLongestLevel(levels) {
+    const lvls = Object.keys(levels).map(level => level.length);
+    return Math.max(...lvls);
+  }
+
+  static calcPadding(level, filler) {
+    const targetLen = Padder.longestLevel + 1 - level.length;
+    const rep = Math.floor(targetLen / filler.length);
+    const padStr = `${filler}${filler.repeat(rep)}`;
+
+    return padStr.slice(0, targetLen);
+  }
+
+  static addPadding(levels, filler = ' ') {
+    Padder.longestLevel = Padder.getLongestLevel(levels);
+    Padder.allPadding = Object.keys(levels).reduce((acc, level) => {
+      acc[level] = Padder.calcPadding(level, filler);
+      return acc;
+    }, {});
+
+    return Padder.allPadding;
+  }
+
+  addPadding(levels, filler) {
+    return Padder.addPadding(levels, filler);
+  }
+
+  transform(info, opts) {
+    info.padding = this.addPadding(opts.levels, opts.filler);
+    return info;
+  }
+}
 
 /*
  * function padLevels (info)
  * Returns a new instance of the padLevels Format which pads
  * levels to be the same length. This was previously exposed as
  * { padLevels: true } to transports in `winston < 3.0.0`.
+ * TODO: Make `opts.levels` default to `winston.config.npm`.
```

Comment:
👍– thanks for calling this out. If you're interested in making this a reality the process would be:

1. Move [cli.js](https://github.com/winstonjs/winston/blob/master/lib/winston/config/cli.js), [npm.js](https://github.com/winstonjs/winston/blob/master/lib/winston/config/npm.js), and [syslog.js](https://github.com/winstonjs/winston/blob/master/lib/winston/config/syslog.js) from `winston` into `triple-beam`.
2. Use that here. 

This is tracked in the [Roadmap](https://github.com/winstonjs/winston/blob/master/CONTRIBUTING.md#core-logging) as _"Move `winston.config` into `triple-beam` around a base `Levels` class."_

Reply:
Yes, I've read the `CONTRIBUTING` file of `winston`, that's why I added the TODO. :wink: